dagshub/data_engine/model/query_result.py

```diff
@@ -401,6 +405,78 @@ def download_binary_columns(
             num_proc=num_proc,
         )
 
+    def predict_with_mlflow_model(
+        self,
+        repo: str,
+        name: str,
+        version: str = "latest",
+        pre_hook: Callable[[Any], Any] = lambda x: x,
+        post_hook: Callable[[Any], Any] = lambda x: x,
+        batch_size: int = 1,
+        log_to_field: Optional[str] = None,
+    ) -> Optional[list]:
+        """
+        Sends all the datapoints returned in this QueryResult as prediction targets for
+        an MLFlow model registered on DagsHub.
+
+        Args:
+            repo: repository to extract the model from
+            name: name of the model in the mlflow registry
+            version: (optional, default: 'latest') version of the model in the mlflow registry
+            pre_hook: (optional, default: identity function) function that runs before datapoint is sent to the model
+            post_hook: (optional, default: identity function) function that converts mlflow model output
+            to the desired format
+            batch_size: (optional, default: 1) function that sets batch_size
+            log_to_field: (optional, default: 'prediction') write prediction results to metadata logged in data engine.
+            If None, just returns predictions.
+            (in addition to logging to a field, iff that parameter is set)
+        """
+
+        # to support depedency-free dataloading, `Batcher` is a barebones dataloader that sets up batched inference
+        class Batcher:
+            def __init__(self, dset, batch_size):
+                self.dset = dset
+                self.batch_size = batch_size
+
+            def __iter__(self):
+                self.curr_idx = 0
+                return self
+
+            def __next__(self):
+                self.curr_idx += self.batch_size
+                return [self.dset[idx] for idx in range(self.curr_idx - self.batch_size, self.curr_idx)]
+
+        prev_uri = os.getenv("MLFLOW_TRACKING_URI", "")
+        os.environ["MLFLOW_TRACKING_URI"] = f"https://dagshub.com/{repo}.mlflow"
+        os.environ["MLFLOW_TRACKING_USERNAME"] = get_token()
+        os.environ["MLFLOW_TRACKING_PASSWORD"] = get_token()
```

Comment:
OK, this still has one problem, which is that this only works with `dagshub.com`.

So what I suggest:
- Assume first that the repo with the model is on the same host as this repo. Get the host from `self.datasource.source.repoApi.host`.
- Use [multi_urljoin()](https://github.com/DagsHub/client/blob/dd3e1919281e5b2ebeaa247fe3ecc9ea615d2f14/dagshub/common/util.py#L11) to construct the TRACKING_URI without having to bother with remembering if there's a slash in the host or not: 
`multi_urljoin(host, f"{repo}.mlflow")`  
- Add a `host` argument to the functions, that allows you to override this behavior, if e.g. I'm on an on-prem installation and want to use a model from regular dagshub.com

P.S. also you can replace the `getenv` with `mlflow.get_tracking_uri()`, but this is a nitpick